pandas/tests/indexing/test_indexing.py

```diff
@@ -846,6 +846,21 @@ def test_non_reducing_slice(self):
             tslice_ = _non_reducing_slice(slice_)
             assert isinstance(df.loc[tslice_], DataFrame)
 
+    def test_non_reducing_slice_on_multiindex(self):
+        dic = {
+            ('a', 'd'): [1, 4],
+            ('a', 'c'): [2, 3],
+            ('b', 'c'): [3, 2],
+            ('b', 'd'): [4, 1]
+        }
+
+        df = pd.DataFrame(dic, index=[0, 1])
+        idx = pd.IndexSlice
+
+        slice_ = idx[:, idx['b', 'd']]
+        tslice_ = _non_reducing_slice(slice_)
+        assert isinstance(df.loc[tslice_], DataFrame)
```

Comment:
can you compute the result here  and construct and expected DataFrame and use ``tm.assert_frame_equal`` to compare)

(and fix the test above as well)

Reply:
can you move this and the test above to test_multiindex.py

Reply:
> (and fix the test above as well)

You mean a list of expected DataFrames when slice in slices would be .loc'ed in `test_non_reducing_slice`?

Reply:
> can you move this and the test above to test_multiindex.py

AFAICS, `test_non_reducing_slice` does not concern MultiIndex, should I move it regardless?